### PR TITLE
Update esbuild to 0.6.3

### PR DESCRIPTION
**Pull request by drwpow**
Also fixes a TypeScript error.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export type Options = {
   exclude?: FilterPattern
   watch?: boolean
   minify?: boolean
-  target?: Target
+  target?: string | string[]
```
Comment (drwpow) [inline src/index.ts:19]: This just copied the type that esbuild uses in [main.d.ts](https://unpkg.com/browse/esbuild@0.6.3/lib/main.d.ts#LC9)

File: test/index.test.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ test('load index.(x)', async () => {
   expect(output[0].code).toMatchInlineSnapshot(`
     "class Foo {
       render() {
-        return React.createElement(\\"div\\", {
+        return /* @__PURE__ */ React.createElement(\\"div\\", {
```
Comment (drwpow) [inline test/index.test.ts:79]: I think this is just from an esbuild version bump. Not sure how to disable this?

File: package.json [hunk 3]
```diff
@@ -28,6 +28,6 @@
   },
   "dependencies": {
     "@rollup/pluginutils": "^3.1.0",
-    "esbuild": "^0.5.9"
+    "esbuild": "^0.6.3"
```
Comment (drwpow) [inline package.json:31]: What are your thoughts on moving this to a `peerDependency`? That way the version of esbuild is less-locked-in to this plugin.

Comment (egoist) [reply package.json:31]: 👌 that's probably better

Comment (egoist) [reply package.json]: you need to install it as devDep so we can run the test

Comment (drwpow) [reply package.json]: Oh shoot 🤦. Sorry. Added!

**Pull request merged**